src/main/java/frc/robot/subsystems/SwerveDrive.java

```diff
@@ -210,7 +231,10 @@ public void periodic() {
   }
 
   public void resetPose(Pose2d robotPose){
-    m_odometry.resetPosition(getRotation(), getModulePositions(), robotPose);
+    if (Robot.isSimulation()) {
+      m_simrotation = robotPose.getRotation();
```

Comment:
This was the important change to get this working in the simulator. 
We need the simulator angle and the odometry angle to match - to do that, we need to make sure the sim angle will be the target angle before we reset the odometry